java-frontend/src/main/java/org/sonar/java/se/checks/RedundantAssignmentsCheck.java

```diff
@@ -97,28 +106,39 @@ public void checkEndOfExecution(CheckerContext context) {
     for (Map.Entry<AssignmentExpressionTree, Collection<Assignment>> assignmentForTree : contexts.pop().asMap().entrySet()) {
       Collection<Assignment> allAssignments = assignmentForTree.getValue();
       if (allAssignments.stream().allMatch(Assignment::isRedundant)) {
-        Assignment assignment = Iterables.getFirst(allAssignments, null);
+        Set<List<Location>> flows = allAssignments.stream().map(Assignment::flows).flatMap(Set::stream).collect(Collectors.toSet());
         reportIssue(assignmentForTree.getKey(),
-          String.format("Remove this useless assignment; \"%s\" already holds the assigned value along all execution paths.", assignment.symbol.name()));
+          String.format("Remove this useless assignment; \"%s\" already holds the assigned value along all execution paths.",
+            Iterables.getFirst(allAssignments, null).assignedSymbol.name()),
+          flows);
       }
     }
   }
 
   private static class Assignment {
 
-    private final Symbol symbol;
+    private final Symbol assignedSymbol;
+    @Nullable
+    private final Symbol fromSymbol;
     private final SymbolicValue oldValue;
     private final SymbolicValue newValue;
+    private final ExplodedGraph.Node node;
 
-    public Assignment(Symbol symbol, SymbolicValue oldValue, SymbolicValue newValue) {
-      this.symbol = symbol;
+    public Assignment(Symbol assignedSymbol, @Nullable SymbolicValue oldValue, SymbolicValue newValue, @Nullable Symbol fromSymbol, ExplodedGraph.Node node) {
```

Comment:
Name of this data class is a bit dodgy... hard to get its just a data class from its name only. Not sure how to refine it though.